Added Bitrise / Fastlane doc (#773)
fastlane/docs: docs/best-practices/continuous-integration/bitrise.md

# Bitrise Integration

[Bitrise](https://www.bitrise.io) allows you to automatically build, test and deploy your iOS and Android apps.

## How to get started?

Using _fastlane_ for your Bitrise workflow is simple. Just add the [_fastlane_](https://www.bitrise.io/integrations/steps/fastlane) step to your
workflow [(guide)](https://devcenter.bitrise.io/getting-started/manage-your-bitrise-workflow),
after the [`Git Clone`](https://www.bitrise.io/integrations/steps/git-clone) step (and any other dependency step).

> When you add the _fastlane_ step, we ensure that you are running on the latest _fastlane_ version as it is pre-installed on all our VMs. Inside the step you can set the _fastlane_ action and we will run it automatically every time you push a new code change.

Reviewer: ```suggestion
When you add the _fastlane_ step, Bitrise ensures that you are running on the latest _fastlane_ version as it is pre-installed on all its VMs. Inside the step you can set the _fastlane_ action and Bitrise will run it automatically every time you push a new code change.
```

# Bitrise Integration

[Bitrise](https://www.bitrise.io) allows you to automatically build, test and deploy your iOS and Android apps.

## How to get started?

> Using _fastlane_ for your Bitrise workflow is simple. Just [add the ](https://devcenter.bitrise.io/getting-started/manage-your-bitrise-workflow) [_fastlane_](https://www.bitrise.io/integrations/steps/fastlane) [step to your

Reviewer: The linking going on here is pretty confusing. Can this be rewritten so these two links have a word in between somehow and a link is not repeated?

(also: additional space after `add the` that should be removed)

workflow](https://devcenter.bitrise.io/getting-started/manage-your-bitrise-workflow),
after the `Git Clone` step (and any other dependency step).

> ### _Have our Certificate and profile installer step in your workflow!_

Reviewer: Can we get rid of the italic here? It makes the whole paragraph a lot harder to read.

Author: Yup, fixed

> _You should also add/keep the `Certificate and profile installer` step in the workflow, to download your .p12 Certificates and Provisioning Profiles uploaded to [bitrise.io](https://www.bitrise.io) and to install them. **Even if you don't upload your files to** [**bitrise.io**](https://www.bitrise.io) **and instead you use a_ _fastlane_ _tool to manage your code signing files you should still keep this step in the workflow**. Read more about [iOS Code Signing using third party tools](https://devcenter.bitrise.io/ios/code-signing/#use-a-third-party-tool-to-manage-your-code-signing-files)._

Reviewer: This line doesn't render properly: https://github.com/fastlane/docs/blob/cf1cd9a565ff9d8ee66e719260bf28f7619029dd/docs/best-practices/continuous-integration/bitrise.md

> With adding the _fastlane_ step we ensure that you are running on the latest _fastlane_ version, as it is pre-installed on all our VMs. Inside the step you can set the _fastlane_ action and we will run it automatically every time you push a new code change.

Reviewer: How is this related to its headline? Should this be above under the first "Getting started"?

> For more configuration options see the `fastlane` step's description in the Workflow Editor!

Reviewer: Can this get context with a link?

### _iOS code signing guide_

> _If you want to use [bitrise.io](https://www.bitrise.io) to store your code signing files, you should just follow the [iOS Code Signing guide here](https://devcenter.bitrise.io/ios/code-signing/)._

Reviewer: Pure fastlane signing (e.g. `match` + `gym`?) does not work?

## How to configure _fastlane match_ for Bitrise

> If you want to use [_fastlane match_](https://github.com/fastlane/fastlane/tree/master/match)

Reviewer: Best add `sync_code_signing` here and link to docs at https://docs.fastlane.tools/actions/sync_code_signing/ and https://docs.fastlane.tools/actions/match/

in your [bitrise.io](https://www.bitrise.io/) build you only have to do three things:

1. Make sure that a single SSH key can be used to `git clone` both your main repository (the one
   you register on [bitrise.io](https://www.bitrise.io/)) and the `match` repository.

>    You can find more info [in this guide](https://github.com/bitrise-io/devcenter/blob/master/faq/adding-projects-with-submodules).

Reviewer: How is this related to submodules? Or does this just accidentally answer the same question?

2. Add an environment variable `MATCH_PASSWORD`, as

>    [described in ](https://github.com/fastlane/fastlane/tree/master/match#encryption-password) [_match_](https://github.com/fastlane/fastlane/tree/master/match#encryption-password) ['s docs](https://github.com/fastlane/fastlane/tree/master/match#encryption-password),

Reviewer: The links here are outdated - this should also link to docs.fastlane.tools (and update your knowledge base as well ;) ). Also no need to have 3 different links.

   to specify the `Encryption password` you used for `match`.

>    On [bitrise.io](https://www.bitrise.io/) you should add this as a `Secret Environment Variable`,

Reviewer: less general links to bitrise maybe? Just say "Bitrise" in the text itself.

   in the [Workflow Editor](http://devcenter.bitrise.io/docs/add-your-first-step-to-your-apps-workflow).
   _Make sure to disable_ the `Replace variables in input?` option of the environment
   variable, to not to cause issues when the value includes the `$` (dollar) sign, which is used
   for environment variable expansion.

3. Make sure to use `match`'s `readonly` mode, or else `match` will try to connect
   to the Apple Developer Portal, which requires further authorization (providing additional
   username and password for Apple Dev Portal login)!
   * If you use `match` in your `Fastfile` or `fastlane` config: `match(app_identifier: "my.domain", type: "appstore", readonly: true)`

>    * If you use it as a command line tool: `match development --readonly`

Reviewer: Can you just run the CLI in Bitrise?

   * More info in `match`'s [official readme / docs](https://docs.fastlane.tools/actions/match/)

> ## What's next?

Reviewer: This section needs a bit more structure. Paragraph, new lines...
